test/service/MarriageAllowanceServiceTest.scala

```diff
@@ -16,23 +16,123 @@
+import connectors.{EmailConnector, MarriageAllowanceDESConnector, MarriageAllowanceDataConnector}
+import errors.TooManyRequestsError
 import metrics.Metrics
 import models.ApiType.ApiType
 import models._
-import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import org.joda.time.LocalDate
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito._
+import org.scalatest.TestData
+import org.scalatest.mockito.MockitoSugar
+import org.scalatestplus.play.guice.GuiceOneAppPerTest
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.libs.json.Json
 import services.MarriageAllowanceService
 import test_utils.TestUtility
+import uk.gov.hmrc.domain.Generator
 import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, _}
 import uk.gov.hmrc.play.test.UnitSpec
 import utils.WSHttp
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.{ExecutionContext, Future}
 
-class MarriageAllowanceServiceTest extends UnitSpec with TestUtility with GuiceOneAppPerSuite {
+//TODO remove the need for TestUtility
+class MarriageAllowanceServiceTest extends UnitSpec with TestUtility with MockitoSugar with GuiceOneAppPerTest {
+
+  implicit override def newAppForTest(testData: TestData): Application = {
+
+    val builder = new GuiceApplicationBuilder()
+
+    testData.name match {
+      case testName if testName.contains("post enabled is true") => builder.configure("des.post.enabled" -> true).build()
+      case testName if testName.contains("post enabled is false") => builder.configure("des.post.enabled" -> false).build()
+      case _ => builder.build()
+    }
+  }
+
+
+  val year = 2019
+  val generatedNino = new Generator().nextNino
+
+   trait Setup {
+     val cID = 123456789
+     val findRecipientRequest = FindRecipientRequest(name = "testForename1", lastName = "testLastName",
+       gender = Gender("M"), nino = generatedNino, dateOfMarriage = Some(new LocalDate(year,12,12)))
+     val userRecord = UserRecord(cid = cID, timestamp = "20200116155359011123")
+
+   }
+
+
+  lazy val service = new MarriageAllowanceService {
+    override val dataConnector = mock[MarriageAllowanceDataConnector]
+    override val emailConnector = mock[EmailConnector]
+    override val metrics = mock[Metrics]
+    override val startTaxYear = START_TAX_YEAR
+    override val maSupportedYearsCount = MA_SUPPORTED_YEARS_COUNT
+
+    override def currentTaxYear: Int = year
+
+  }
+
+  "MarriageAllowanceService" should {
+
+    "return a MarriageAllowanceDESConnector if toggle post enabled is true" in {
+      MarriageAllowanceService.getConnectorImplementation shouldBe MarriageAllowanceDESConnector
+    }
+
+    "return a MarriageAllowanceDataConnector if toggle post enabled is false" in {
+      MarriageAllowanceService.getConnectorImplementation shouldBe MarriageAllowanceDataConnector
+    }
+  }
+
+
+  "getRecipientRelationship" should {
+
+    "return a valid response" in { new Setup {
```

Comment:
More info in test scenario setup for both of these please to describe the behaviour